File: perfect-square/perfect_square.rb

def perfect_square(number)
  perfect=false
  x=1
  current=1

  while not perfect and current < number
    current=x*x
    perfect= current == number
    x+=1
  end
  return perfect
end

puts("Enter a number")
n=gets.chomp
is_perfect=perfect_square(n.to_i)
if is_perfect
  puts("It is a perfect square")
else
  puts("It is not a perfect square")
end

File: perfect-square/perfect-square.cpp
#include <iostream>
#include <cmath>

using namespace std;

bool perfect_square(int num)
{
    float temp = sqrt(num);

    if (temp - (int)sqrt(num) == 0)
        return true;

    else return false;
}

int main()
{
    cout << "Enter a number to test!: ";
    int num;
    cin >> num;
    
    if (perfect_square(num))
    {
        cout << num << " is a perfect square!";
    }
    else
    {
        cout << num << " is not a perfect square!";
    }
}

